BurstBotNET/Commands/Trade/Exchange.cs

```diff
@@ -0,0 +1,96 @@
+using System.ComponentModel;
+using System.Globalization;
+using BurstBotShared.Api;
+using BurstBotShared.Shared;
+using BurstBotShared.Shared.Extensions;
+using BurstBotShared.Shared.Models.Data.Serializables;
+using BurstBotShared.Shared.Utilities;
+using Flurl.Http;
+using Microsoft.Extensions.Logging;
+using Remora.Commands.Attributes;
+using Remora.Discord.Extensions.Embeds;
+using Remora.Results;
+
+namespace BurstBotNET.Commands.Trade;
+
+public partial class Trade
+{
+    [Command("exchange")]
+    [Description("Exchange for lottery credits with your own tips. 10 tips = 1 credit.")]
+    public async Task<IResult> Exchange(
+        [Description("The amount of tips you want to exchange for lottery credits. 10 tips = 1 credit.")]
+        int amount)
+    {
+        var member = await Utilities.GetUserMember(_context, _interactionApi,
+            "Sorry, but you can only exchange for credits in a guild!", _logger);
+
+        if (member == null) return Result.FromSuccess();
+
+        await _state.AuthenticationService.Login();
+        var (isValid, invokerTip) = await Game.ValidatePlayers(
+            _context.User.ID.Value,
+            new[] { _context.User.ID.Value },
+            _state.BurstApi,
+            _context,
+            10.0f,
+            _interactionApi);
+
+        if (!isValid || invokerTip == null)
+        {
+            var editResult = await _interactionApi
+                .EditOriginalInteractionResponseAsync(_context.ApplicationID, _context.Token,
+                    "Sorry, but you don't have enough tips to exchange!");
+            return editResult.IsSuccess ? Result.FromSuccess() : Result.FromError(editResult);
+        }
+
+        var payload = new
+        {
+            Credit = amount / 10,
+            ChannelId = "0"
+        };
```

Comment:
I didn't know you can do this in C#, it's not clear what it constructs tho

Reply:
It's an anonymous struct because this payload will only be used here so I think it isn't really worth creating a dedicated record type for it. The name of the variable is already `payload`.